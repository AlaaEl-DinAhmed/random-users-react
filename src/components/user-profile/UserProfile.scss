.user {
  border: 1px solid #eee;
  display: flex;
  align-items: center;
  justify-content: center;
  min-height: 100px;
  max-width: 290px;
  &--avatar {
    height: 80px;
    width: 80px;
    border-radius: 60%;
  }
  &--name {
    margin-inline-start: 10px;
  }
}
