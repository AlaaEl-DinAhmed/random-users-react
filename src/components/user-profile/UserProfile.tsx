import User from 'interfaces/User';
import setUserName from 'utils/setUserName';

const UserProfile: React.FC<{ user: User }> = ({ user }) => {
  return (
    <section>
      <img src={user.picture.medium} alt={user.name.first} />
      <p>{setUserName(user.name.first, user.name.last)}</p>
    </section>
  );
};

export default UserProfile;
