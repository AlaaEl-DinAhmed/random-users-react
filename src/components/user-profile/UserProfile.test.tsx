import { render, screen } from "@testing-library/react";
import IUser from "interfaces/User";
import UserProfile from "./UserProfile";

describe("UserProfile Component", () => {
  const user: IUser = {
    name: {
      first: "Alaa",
      last: "Ahmad",
    },
    picture: {
      large: "https://via.placeholder.com/140x100",
      medium: "https://via.placeholder.com/140x100",
      thumbnail: "https://via.placeholder.com/140x100",
    },
  };

  it("should render user image", () => {
    render(<UserProfile user={user} />);

    const img = screen.getByRole("img");

    expect(img).toHaveAttribute("src", user.picture.medium);
  });

  it("should render user name", () => {
    render(<UserProfile user={user} />);

    const pElm = screen.getByText("Alaa Ahmad");

    expect(pElm).toBeInTheDocument();
  });
});
