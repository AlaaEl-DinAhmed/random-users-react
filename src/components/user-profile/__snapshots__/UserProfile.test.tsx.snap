// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Component render userProfile component 1`] = `
<div>
  <div
    class="user"
  >
    <img
      alt="Alaa"
      class="user--avatar"
      src="https://via.placeholder.com/140x100"
    />
    <p
      class="user--name"
    >
      Alaa Ahmad
    </p>
  </div>
</div>
`;
